vividus-plugin-db/src/main/java/org/vividus/bdd/transformer/DatabaseTableTransformer.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.transformer;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Named;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jbehave.core.model.ExamplesTable.TableProperties;
+import org.jbehave.core.model.TableParsers;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.vividus.bdd.db.DataSourceManager;
+import org.vividus.bdd.util.ExamplesTableProcessor;
+
+@Named("FROM_DB")
+public class DatabaseTableTransformer implements ExtendedTableTransformer
+{
+    private final DataSourceManager dataSourceManager;
+
+    public DatabaseTableTransformer(DataSourceManager dataSourceManager)
+    {
+        this.dataSourceManager = dataSourceManager;
+    }
+
+    @Override
+    public String transform(String tableAsString, TableParsers tableParsers, TableProperties properties)
+    {
+        checkTableEmptiness(tableAsString);
+        String dbKey = properties.getMandatoryNonBlankProperty("dbKey", String.class);
+        String sqlQuery = properties.getMandatoryNonBlankProperty("sqlQuery", String.class);
+        List<Map<String, Object>> result = dataSourceManager.getJdbcTemplate(dbKey).queryForList(sqlQuery);
+        if (result.isEmpty())
+        {
+            throw new EmptyResultDataAccessException("Result was expected to have at least one row", 1);
+        }
+        return ExamplesTableProcessor.buildExamplesTable(
+                result.get(0).keySet(),
+                result.stream()
+                        .map(map -> map
+                                .entrySet()
+                                .stream()
+                                .collect(toMap(
+                                        Map.Entry::getKey,
+                                        e -> Optional.ofNullable(e.getValue())
```

Comment:
Is it expected by the feature requester to tread a null as empty?  We have a similar case where null should be null

Reply:
Discussed with Anton P. to treat null as empty for now.

Reply:
2. Should I distinguish them now for some reason/use case? What is the `null` for Examples Tables? What code changes expected?

Reply:
2.1. I don't know your case in details, but as it was mentioned before it's not relevant for you. But since it's a transformer that could be used by other teams, we should make it generic as much as possible. If we skip `null`-handling now, it'll lead to breaking changes in future, so everyone will be affected
2.2. https://docs.vividus.dev/vividus/latest/tests-configuration.html#_mapping_values_to_null_s
2.3. I think we should remove conversion of `null`-s to empty strings by default, because it's a lossy conversion. As an option is to add new transformer property, e.g. `nullReplacement= `